Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -378,6 +388,49 @@ EncodedDataStatus ImageDecoderCG::encodedDataStatus() const
     return m_encodedDataStatus;
 }
 
+bool ImageDecoderCG::hasHDRGainMapContent() const
+{
+#if HAVE(SUPPORT_HDR_DISPLAY)
+    auto properties = adoptCF(CGImageSourceCopyProperties(m_nativeDecoder.get(), createImageSourceMetadataOptions().get()));
+    if (!properties)
+        return false;
+
+    auto fileContentsProperties = (CFDictionaryRef)CFDictionaryGetValue(properties.get(), kCGImagePropertyFileContentsDictionary);
```

Comment:
Should we validate that the value is a CFDictionary?

Reply:
c